Maple2.Model/Metadata/BeginCondition.cs

```diff
@@ -7,10 +8,19 @@ public record BeginCondition(
     long Mesos,
     Gender Gender,
     JobCode[] JobCode,
+    float Probability,
+    bool OnlyShadowWorld,
+    bool OnlyFlyableMap,
+    Dictionary<BasicAttribute, long> Stat,
+    BeginConditionWeapon[]? Weapon,
     BeginConditionTarget? Target,
     BeginConditionTarget? Owner,
     BeginConditionTarget? Caster);
 
+public record BeginConditionWeapon(
```

Comment:
curious what these bytes even mean, should it be an enum maybe?

Reply:
You sort of have it laid out in `ItemType` record struct. I believe the lua files use these values too.
We can definitely do it for equips/weapons but everything else I'm not sure about (e.g. ObjectWeapon can be a value between 80 and 89). There is also the case where furnishing blocks start to collide with other type #s.
We can do a `EquipType` enum if that works?

Reply:
Would it make sense to just use that struct here instead? (With `Group` hardcoded as 1)